pkg/api/org_users.go

```diff
@@ -386,6 +390,17 @@ func (hs *HTTPServer) updateOrgUserHelper(c *contextmodel.ReqContext, cmd org.Up
 	if !c.OrgRole.Includes(cmd.Role) && !c.IsGrafanaAdmin {
 		return response.Error(http.StatusForbidden, "Cannot assign a role higher than user's role", nil)
 	}
+	if hs.Features.IsEnabled(featuremgmt.FlagOnlyExternalOrgRoleSync) {
+		// we do not allow to change role for external synced users
+		qAuth := login.GetAuthInfoQuery{UserId: cmd.UserID}
+		err := hs.authInfoService.GetAuthInfo(c.Req.Context(), &qAuth)
+		if err != nil {
+			return response.Error(http.StatusInternalServerError, "Failed to get user auth info", err)
+		}
+		if qAuth.Result.AuthModule != "" && login.IsExternallySynced(hs.Cfg, qAuth.Result.AuthModule) {
```

Comment:
`IsExternallySynced` is using Auth Label to check settings while here it's being provided AuthModule.

Backend check will always return false here.

Change `IsExternallySynced` to always use the AuthModule and not the frontend only Auth Label